Mathlib/Topology/DerivedSet.lean

```diff
@@ -0,0 +1,101 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.Topology.Perfect
+
+/-!
+# Derived set
+
+This file defines the derived set of a set, the set of all `AccPt`s of its principal filter,
+and proves some properties of it.
+
+-/
+
+open Filter Topology
+
+variable {X : Type*} [TopologicalSpace X]
+
+theorem AccPt.map {β : Type*} [TopologicalSpace β] {F : Filter X} {x : X}
+    (h : AccPt x F) {f : X → β} (hf1 : ContinuousAt f x) (hf2 : Function.Injective f) :
+    AccPt (f x) (map f F) := by
+  apply map_neBot (m := f) (hf := h) |>.mono
+  rw [Filter.map_inf hf2]
+  gcongr
+  apply tendsto_nhdsWithin_of_tendsto_nhds_of_eventually_within _ hf1.continuousWithinAt
+  simpa [hf2.eq_iff] using eventually_mem_nhdsWithin
+
+/--
+The derived set of a set is the set of all accumulation points of it.
+-/
+def derivedSet (A : Set X) : Set X := {x | AccPt x (𝓟 A)}
```

Comment:
I wonder if there's any reason *not* to allow for arbitrary filters as inputs. I guess it means you can't iterate the `derivedSet` as easily.

Reply:
The theorems would be less pretty, and if you're using an arbitrary filter you could just directly use `AccPt` — how often do you want it as a set?